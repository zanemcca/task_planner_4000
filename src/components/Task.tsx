
import React from 'react';
import {
  Col,
  Row,
  Card,
  Icon
} from 'antd'
import moment from 'moment';

import '../styles/Task.css';
import { IconProps } from 'antd/lib/icon';
import Channel, { IChannel } from './Channel';

export interface ITask {
  description: string
  time?: moment.Duration
  channel?: IChannel
  complete?: boolean
}

export type ITaskProps = ITask

const Task = (props: ITaskProps) => {
  const taskCompleteIconProps: IconProps = {
    type: 'check-circle',
    style: {
      height: 20,
      width: 20
    }
  }

  if (props.complete) {
    taskCompleteIconProps.theme = 'twoTone'
    taskCompleteIconProps.twoToneColor = '#52c41a'
  }

  return (
    <Card className="Task">
      <Row className="Task-header">
        <div className="Task-description">{props.description}</div>
        {props.time && <div className="Task-time">{props.time.humanize()}</div>}
      </Row>
      <Row className="Task-footer">
        <Col span={3} className="Task-complete">
          <Icon {...taskCompleteIconProps} />
        </Col>
        <Col span={21} className="Task-channel">{props.channel && <Channel {...props.channel}/>}</Col>
      </Row>
    </Card>
  )
}

export default Task;
