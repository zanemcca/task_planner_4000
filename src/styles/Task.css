.Task {
}

.Task-header {
  display: flex
}

.Task-footer {
}

.Task-description {
  text-align: left;
  font-family: Avenir;
  font-style: normal;
  font-weight: 500;
  font-size: 14px;
  line-height: 19px;
  flex: 1 1 100%;
}

.Task-time {
  background: #F0F0F0;
  border-radius: 4px;
  font-family: Avenir;
  font-style: normal;
  font-weight: 500;
  font-size: 8px;
  line-height: 19px;

  padding-left: 5px;
  padding-right: 5px;

  flex: 0 0 auto;

  color: #787878;
}

.Task-channel {
  text-align: right;
  margin-top: 9px;
}

.Task-complete {
  font-size: 20px;
  text-align: left;
}
