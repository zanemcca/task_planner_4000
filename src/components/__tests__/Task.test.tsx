import renderer from 'react-test-renderer';

import Task, { ITaskProps } from '../Task';
import moment from 'moment';
import React from 'react';

it('should render in a nice way', () => {
  const props: ITaskProps = {
    description: 'Aliquam vel massa accumsan, accumsan ex eu, posuere nunc.',
    complete: true,
    time: moment.duration('00:30:00'),
    channel: {
      color: 'blue',
      title: 'important'
    }
  }
  const component = renderer.create( <Task {...props}/>)
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})
