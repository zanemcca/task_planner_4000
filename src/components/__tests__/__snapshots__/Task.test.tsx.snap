// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match the snapshot 1`] = `
<div
  className="ant-card Task ant-card-bordered"
>
  <div
    className="ant-card-body"
    style={Object {}}
  >
    <div
      className="ant-row Task-header"
      style={Object {}}
    >
      <div
        className="Task-description"
      >
        Aliquam vel massa accumsan, accumsan ex eu, posuere nunc.
      </div>
      <div
        className="Task-time"
      >
        30 minutes
      </div>
    </div>
    <div
      className="ant-row Task-footer"
      style={Object {}}
    >
      <div
        className="ant-col ant-col-3 Task-complete"
        style={Object {}}
      >
        <i
          aria-label="icon: check-circle"
          className="anticon anticon-check-circle"
          style={
            Object {
              "height": 20,
              "width": 20,
            }
          }
        >
          <svg
            aria-hidden="true"
            className=""
            data-icon="check-circle"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
              fill="#52c41a"
            />
            <path
              d="M512 140c-205.4 0-372 166.6-372 372s166.6 372 372 372 372-166.6 372-372-166.6-372-372-372zm193.4 225.7l-210.6 292a31.8 31.8 0 0 1-51.7 0L318.5 484.9c-3.8-5.3 0-12.7 6.5-12.7h46.9c10.3 0 19.9 5 25.9 13.3l71.2 98.8 157.2-218c6-8.4 15.7-13.3 25.9-13.3H699c6.5 0 10.3 7.4 6.4 12.7z"
              fill="#f6ffed"
            />
            <path
              d="M699 353h-46.9c-10.2 0-19.9 4.9-25.9 13.3L469 584.3l-71.2-98.8c-6-8.3-15.6-13.3-25.9-13.3H325c-6.5 0-10.3 7.4-6.5 12.7l124.6 172.8a31.8 31.8 0 0 0 51.7 0l210.6-292c3.9-5.3.1-12.7-6.4-12.7z"
              fill="#52c41a"
            />
          </svg>
        </i>
      </div>
      <div
        className="ant-col ant-col-21 Task-channel"
        style={Object {}}
      >
        <div>
          <span
            style={
              Object {
                "color": "blue",
              }
            }
          >
            # 
          </span>
          <span>
            important
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
